arbor/mechcat.cpp

```diff
@@ -605,4 +624,34 @@ const mechanism_catalogue& load_catalogue(const std::string& fn) {
     return *((const mechanism_catalogue*)get_catalogue());
 }
 
+mechanism_info get_mech_info(const arb_mechanism_type& mech_) {
```

Comment:
Ok, since we are making this PR even more invasive, how about converting `mechanism_info` to a thin wrapper around `arb_mechanism_type`

Reply:
That's indeed the sort of thing I was aiming at with the converting constructor. It could be a thin user friendly layer, or it could copy the data and make the maps etc. Either way works!